docs/usage/configuration/config_documentation.md

```diff
@@ -904,6 +904,23 @@ Example configuration:
 redaction_retention_period: 28d
 ```
 ---
+---
+### `purge_retention_period`
+
+How long to keep locally forgotten room in the DB. After this period the room
+will be fully purged from the DB.
+
+Synapse will check whether the retention period has concluded for room
+purges every hour. Thus, even if this option is set to `0`, Synapse may
+still take up to one hour to purge forgotten rooms from the database.
+
+Defaults to `7d`. Set to `null` to disable.
+
+Example configuration:
+```yaml
+purge_retention_period: 28d
+```
+---
```

Comment:
What worker will the scheduled purges happen on?

Do we need to worry about multiple workers trying to do the same work?

Reply:
I think it's worse than that: we have an in memory `pagination_lock`, and I am pretty sure it just doesn't work correctly in a worker env.
I am expecting `get_messages` to be called on client reader workers while the purge would be triggered on the main process.
I think we need to both pin this to a worker and use the newly introduced in-DB lock to fix current behavior.